programs/lfx-mentorship/2023/03-Sep-Nov/project_ideas.md

```diff
@@ -30,6 +30,27 @@
   - Kevin Hannon, @kannon92, kevin@gr-oss.io
 - Upstream Issue (URL): https://github.com/armadaproject/armada/issues/2702
 
+### CRI-O
+
+#### Add additional log drivers to conmon-rs
+
+- Description: conmon-rs is a container monitor written in Rust, used by CRI-O to monitor a container's lifecycle. Part of its responsibilities is log forwarding--taking the output of the container and writing that output to various places. Currently, conmon-rs supports one format: the one required by the Kubernetes CRI. The goal of this proposal is to add new log formats from the list of standardized ones, like JSON, Splunk, Journald.
+- Expected outcome: A JSON log driver and Journald log driver are added to conmon-rs. A stretch goal of adding a Splunk log driver is also within scope.
+- Recommended Skills: Rust, familiarity with containers
+- Mentor(s):
+  - Peter Hunt, haircommander, pehunt@redhat.com
+  - Sascha Grunert, saschagrunert, sgrunert@redhat.com
+- Upstream Issue (URL): https://github.com/containers/conmon-rs/issues/1126
```

Comment:
I'm concerned that this may not fit the criteria for a CNCF LFX Mentoring project as I don't see _Containers_ as a CNCF project (that said, if it's a new one that I've missed, please accept my mea culpa).

I see that this would help CRI-O; is there a way you could work this into your project more directly, @haircommander?

Reply:
so conmon-rs is a project shared by cri-o and it's sibling (non cncf project) podman. adding support here will directly benefit cri-o as there have been asks of splunk logging in the past, as well as current support for journald logs that needs porting. FWIW: the GSoC project we're currently working on is also on conmon-rs by extension of cri-o

Reply:
Thanks for the context. I think we can bring this in. 🙂